karabo_data/geometry2.py

```diff
@@ -335,6 +335,68 @@ def to_distortion_array(self):
 
         return distortion
 
+    def _tile_slice(self, tileno):
+        """Implement in subclass: which part of module array each tile is.
+        """
+        raise NotImplementedError
+
+    def get_pixel_positions(self, centre=True):
+        """Get the coordinates of each pixel in the detector
+
+        The output is an array with shape like the data, with an extra dimension
+        of length 3 to hold (x, y, z) coordinates. Coordinates are in pixel
+        units from the detector centre.
+
+        If centre=True, the coordinates are calculated for the centre of each
+        pixel. If not, the coordinates are for the first corner of the pixel
+        (the one nearest the [0, 0] corner of the tile in data space.
+        """
+        out = np.zeros(self.expected_data_shape + (3,), dtype=np.float32)
+
+        # Prepare some arrays to use inside the loop
+        pixel_ss_index, pixel_fs_index = np.meshgrid(
+            np.arange(0, self.frag_ss_pixels),
+            np.arange(0, self.frag_fs_pixels),
```

Comment:
You might want to add dtypes to the np.arange calls to avoid incompatible type errors

Reply:
Yep, good call.